core/src/autogluon/core/utils/time.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import time
+from typing import Optional
+
+from pandas import DataFrame
+
+logger = logging.getLogger(__name__)
+
+
+def time_func(f, args: list = None, kwargs: dict = None, time_limit: float = 0.2, max_repeats: int = 10) -> float:
```

Comment:
We should use python's [`timeit()`](https://docs.python.org/3/library/timeit.html#basic-examples):
```python
import timeit
time = timeit.timeit(lambda: f(x), number=100)
```

Reply:
Added TODO. I don't want to use timeit without further testing as time_func works in a very particular way to avoid wasting too much time and I'd like to avoid too much fixed time-sink into estimating time calls. Not urgent, so we can consider post v0.4.0